code/datums/elements/glass_bleeder.dm

```diff
@@ -0,0 +1,27 @@
+/// Drop glass shards when taking damage
+/datum/element/glass_bleeder
+	element_flags = ELEMENT_DETACH_ON_HOST_DESTROY
+
+/datum/element/glass_bleeder/Attach(atom/movable/target)
+	. = ..()
+
+	if(!isliving(target))
+		return ELEMENT_INCOMPATIBLE
+
+	RegisterSignal(target, COMSIG_MOB_APPLY_DAMAGE, PROC_REF(on_apply_damage))
+
+/datum/element/glass_bleeder/Detach(datum/source, ...)
+	UnregisterSignal(source, COMSIG_MOB_APPLY_DAMAGE )
+	return ..()
+
+/datum/element/glass_bleeder/proc/on_apply_damage(mob/living/liver, amount, damage_type)
+	SIGNAL_HANDLER
+
+	if(damage_type != BRUTE)
+		return
+
+	if(amount > 20)
```

Comment:
This one is so close to being a useful generic component, can we pass the spawned paths as args and rename this so nobody makes a virtually identical element later for a different type of debris?